### ISSUE TITLE
iOS: This in-app purchase has already been bought for some sku's

### ISSUE THREAD

**Issue by nohaapav** [dooboolab/react-native-iap#571]
### Version of react-native-iap

2.4.8

### Version of react-native

0.55.4

### Platforms you faced the error (IOS or Android or both?)

iOS 12.x +

### Expected behavior

Dialog saying that purchase was already done, buy for free as for other sku's that have been purchased in past. 

### Actual behavior

For random sku i get following notification dialog:
`This in-app purchase has already been bought.`

### Tested environment (Emulator? Real Device?)

Real device -> iPhone SE, iOS 12.3.1 Sandbox

### Steps to reproduce the behavior

Just call buyProduct for [Non-Consumable] item that was already bought. For random one i get dialog above. There is not even callback so i can do some action like getPurchases.

Is there some work-around to handle this behavior ?

**Comment by hyochan** [comment]
It's a bit confusing for me to understand the current problem. Could you provide more information to support my understanding?

**Comment by nohaapav** [comment]
Sure.

I have multiple products (Non-Consumable). These have been already purchased (everything worked). Let's say i will switch the device and do not restore purchases in first place (`getAvailablePurchases`). When user click buy button so `buyProduct` is called for purchased item i get purchase confirmation dialog. After entering valid password i see 2 possible flows:

1. User get notification dialog saying `This in-app purchase has already been bought`
2. User get confirmation dialog to purchase item for free as it was already bought.

My question is why i can face 2 different flows for purchased product ?

**Comment by hyochan** [comment]
Hmm.. interesting. Thanks for sharing. Let's see who can answer this question.

**Comment by Stafox** [comment]
It might be connected with receipt actuality.

The First dialog might be appear when there is no information about this purchase in some localStorage. And this info become available only after communication with iTunes.
The Second dialog appears when device already knows about this purchase (some info exists in local storage).
